todo-app/src/App.js

```diff
@@ -1,13 +1,98 @@
 import React, { Component } from 'react';
+import Task from './Task' 
+import TaskForm from './TaskForm'
+import {loadTodoItems, saveTodoItems} from './storage'
+
+
+export default class App extends Component {
+    state = {
+    todos: loadTodoItems(),
+    editingTodoID: null
+  }
+
+handleEditTodo = id => {
+  this.setState({editingTodoID: id});
+};
+
+handleCancleEdit = ()=> {
+    this.setState({editingTodoID: null});
+}
+
+
+setTodos(todos) {
+    this.setState({todos})
+    saveTodoItems(todos)
+}
+
+handleSaveEdit = (id , description) => {
+    this.setTodos(this.state.todos.map(todoItem => {
+        if (todoItem.id === id) {
+            this.setState({editingTodoID: null});
+            return {...todoItem, description};
+        }else {
+            return todoItem;
+        }
+    }))
+};
+toggleStatus = (todoID) =>{
+    this.setTodos(this.state.todos.map(todo =>{
+                if (todo.id === todoID){
+                    todo.done = !todo.done
+                }
+                return todo
+                
+            })
+      )
+    }
+
+
+ newTaskWithText(text){
+     const ids = this.state.todos.map(todo => todo.id)
+     return {
+         id: ids.length === 0 ? 1 : Math.max(...ids) + 1,
+         description: text,
+         done: false
+     };
+ }
+ handleTaskAdd = newTodo => {
+     const todo = this.newTaskWithText(newTodo);
+     this.setTodos([todo, ...this.state.todos]);
+ };
+
+handleRemoveTodo = (todoItem) => {
+     const updateTodos = this.state.todos;
+    updateTodos.splice(updateTodos.indexOf(todoItem), 1)
```

Comment:
It would be much more robust to have a `handleRemoveTodo` function that simply took a todo's ID and removed it via `filter`. What you're doing here is actually forcefully mutating your state before even calling `setState`, a React no-no.

Even though you are doing `const updateTodos = this.state.todos`, you're not actually creating a new array anywhere, and `splice` modifies an array in place instead of returning a new array.